src/odata-to-abstract-sql.ts

```diff
@@ -1726,6 +1727,71 @@ export class OData2AbstractSQL {
 			);
 		}
 	}
+	AddJoins(
+		query: Query,
+		parentResource: Resource,
+		match:
+			| GenericPropertyPath<PropertyPath>
+			| Array<GenericPropertyPath<PropertyPath>>,
+		joinType: JoinType,
+	) {
+		if (Array.isArray(match)) {
+			match.forEach((v) => {
+				this.AddJoins(query, parentResource, v, joinType);
+			});
+		} else {
+			let nextProp = match;
+			let prop;
+			while (
+				// tslint:disable-next-line:no-conditional-assignment
+				(prop = nextProp) &&
+				prop.name &&
+				prop.property?.name
+			) {
+				nextProp = prop.property;
+				const resourceAlias = this.resourceAliases[prop.name];
+				if (resourceAlias) {
+					parentResource = resourceAlias;
+				} else {
+					parentResource = this.AddJoinNavigation(
+						query,
+						parentResource,
+						prop.name,
+						joinType,
+					);
+				}
+			}
+			const maybeMethodCallProp = nextProp as { args?: any[] } | undefined;
+			if (maybeMethodCallProp?.args != null) {
+				this.AddJoins(
+					query,
+					parentResource,
+					maybeMethodCallProp.args,
```

Comment:
Heads-up that this was props.args previously:
https://github.com/balena-io-modules/odata-to-abstract-sql/blob/df845be23124cd790ced0526017092c288dc9ed3/src/odata-to-abstract-sql.ts#L1118-L1120
even in the ometajs version:
https://github.com/balena-io-modules/odata-to-abstract-sql/blob/7989ddf8376de7e8b6d770a1782ee2a762e67294/odata-to-abstract-sql.ometajs#L756-L758
but I couldn't think of a reason to not use the `extraProps.args` that we just checked in the `if`, so at the end I changed it.